src/storage/mvcc/reader/forward_seeker.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use kvproto::kvrpcpb::IsolationLevel;
+
+use raftstore::store::engine::IterOption;
+use storage::mvcc::write::{Write, WriteType};
+use storage::mvcc::Result;
+use storage::{Cursor, Key, ScanMode, Snapshot, Statistics, Value, CF_DEFAULT, CF_LOCK, CF_WRITE};
+
+pub struct ForwardSeekerBuilder<S: Snapshot> {
```

Comment:
Please add verbose comments (you can look at the comment style of existing ones). One of our goal of the refinement is to be 100% documentation coverage.